corehq/apps/reports/util.py

```diff
@@ -471,3 +472,17 @@ def resync_case_to_es(domain, case):
         publish_case_saved(case)
     else:
         CommCareCase.get_db().save_doc(case._doc)  # don't just call save to avoid signals
+
+
+def verify_location_allowed_for_case(case, domain, user):
+    from corehq.apps.locations.permissions import user_can_access_location_id, user_can_access_other_user
+    from corehq.apps.reports.standard.cases.data_sources import CaseInfo
+
+    info = CaseInfo(None, case)
+    if info.owner_type == 'location':
+        return user_can_access_location_id(domain, user, info.owner_id)
+    elif info.owner_type == 'user':
+        owning_user = CouchUser.get_by_user_id(info.owner_id)
+        return user_can_access_other_user(domain, user, owning_user)
+    else:
+        return False
```

Comment:
I like the idea of pulling this out into a reusable function.  It might fit better in `locations/permissions.py`, though.  There are a several `user_can_access_X` functions in there, and this would work well as `user_can_access_case` (maybe with the params rearranged to `domain, user, case` for consistency)